### PR TITLE
P4-2690 adding logging to controllers to aid with identifying potential errors and some small refactorings.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
Changes:

- Adding logging to the controllers to aid with error identification in the logs in terms of what the user was trying to do at point of failure.
- Some small refactorings to clean up the controllers interaction with the model e.g. extension functions.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/HtmlController.kt [hunk 1]
```diff
@@ -60,23 +65,29 @@ class HtmlController(@Autowired val moveService: MoveService, @Autowired val jou
...
   fun chooseSupplierGeoAmey(model: ModelMap): RedirectView {
+    logger.info("chosen supplier GEOAmey")
+
     model.addAttribute(SUPPLIER_ATTRIBUTE, Supplier.GEOAMEY)
     return RedirectView(DASHBOARD_URL)
   }
 
   @RequestMapping("$MOVES_BY_TYPE_URL/{moveTypeString}")
-  fun movesById(
+  fun movesByType(
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/HtmlController.kt:83]: Method name was wrong (though technically it has no impact on behavior as it uses the URL.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/HtmlController.kt [hunk 2]
```diff
@@ -139,11 +153,12 @@ class HtmlController(@Autowired val moveService: MoveService, @Autowired val jou
...
-    requestParamStartOfMonth?.let {
-      model.addAttribute(DATE_ATTRIBUTE, requestParamStartOfMonth)
-    }
-    val startOfMonth = model.getAttribute(DATE_ATTRIBUTE) as LocalDate
-    val endOfMonth = endOfMonth(startOfMonth)
+    logger.info("dashboard for $supplier")
+
+    requestParamStartOfMonth?.let { model.addAttribute(DATE_ATTRIBUTE, requestParamStartOfMonth) }
+
+    val startOfMonth = model.getStartOfMonth()
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/HtmlController.kt:160]: Using inline functions to aid readability.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/HtmlController.kt [hunk 3]
```diff
@@ -259,27 +274,33 @@ class HtmlController(@Autowired val moveService: MoveService, @Autowired val jou
...
-      return "no-search-journeys-results"
+      "no-search-journeys-results"
     } else {
       model.addAttribute("journeys", journeys)
-      return "search-journeys-results"
+      "search-journeys-results"
     }
   }
 
+  private fun ModelMap.getStartOfMonth() =
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/HtmlController.kt:297]: New inline functions as part of refactoring.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/HtmlController.kt [hunk 4]
```diff
@@ -199,6 +218,7 @@ class HtmlController(@Autowired val moveService: MoveService, @Autowired val jou
     model: ModelMap,
     redirectAttributes: RedirectAttributes
   ): String {
+    logger.info("")
```
Comment (Thource) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/HtmlController.kt:221]: Empty log msg?

**Review by Thource — APPROVED** [review]

**Pull request merged**